tensorly/backend/core.py

```diff
@@ -729,82 +726,67 @@ def partial_svd(self, matrix, n_eigenvecs=None, random_state=None, **kwargs):
+            v0 = random_state.uniform(-1, 1, min_dim)
+
+        # First choose whether to use X * X.T or X.T *X
+        if dim_1 < dim_2:
+            S, U = scipy.sparse.linalg.eigsh(
+                np.dot(matrix, matrix.T.conj()), k=n_eigenvecs, which='LM', v0=v0
+            )
+            S = np.where(np.abs(S) <= np.finfo(S.dtype).eps, 0, np.sqrt(S))
+            V = np.dot(matrix.T.conj(), U * np.where(np.abs(S) <= np.finfo(S.dtype).eps, 0, 1/S)[None, :])
         else:
-            # We can perform a partial SVD
-            # construct np.random.RandomState for sampling a starting vector
-            if random_state is None:
-                # if random_state is not specified, do not initialize a starting vector
-                v0 = None
-            elif isinstance(random_state, int):
-                rns = np.random.RandomState(random_state)
-                # initilize with [-1, 1] as in ARPACK
-                v0 = rns.uniform(-1, 1, min_dim)
-            elif isinstance(random_state, np.random.RandomState):
-                # initilize with [-1, 1] as in ARPACK
-                v0 = random_state.uniform(-1, 1, min_dim)
-
-            # First choose whether to use X * X.T or X.T *X
-            if dim_1 < dim_2:
-                S, U = scipy.sparse.linalg.eigsh(
-                    np.dot(matrix, matrix.T.conj()), k=n_eigenvecs, which='LM', v0=v0
-                )
-                S = np.where(np.abs(S) <= np.finfo(S.dtype).eps, 0, np.sqrt(S))
-                V = np.dot(matrix.T.conj(), U * np.where(np.abs(S) <= np.finfo(S.dtype).eps, 0, 1/S)[None, :])
-            else:
-                S, V = scipy.sparse.linalg.eigsh(
-                    np.dot(matrix.T.conj(), matrix), k=n_eigenvecs, which='LM', v0=v0
-                )
-                S = np.where(np.abs(S) <= np.finfo(S.dtype).eps, 0, np.sqrt(S))
-                U = np.dot(matrix, V) *  np.where(np.abs(S) <= np.finfo(S.dtype).eps, 0, 1/S)[None, :]
-
-            # WARNING: here, V is still the transpose of what it should be
-            U, S, V = U[:, ::-1], S[::-1], V[:, ::-1]
-            V = V.T.conj()
+            S, V = scipy.sparse.linalg.eigsh(
+                np.dot(matrix.T.conj(), matrix), k=n_eigenvecs, which='LM', v0=v0
+            )
+            S = np.where(np.abs(S) <= np.finfo(S.dtype).eps, 0, np.sqrt(S))
+            U = np.dot(matrix, V) *  np.where(np.abs(S) <= np.finfo(S.dtype).eps, 0, 1/S)[None, :]
+
+        # WARNING: here, V is still the transpose of what it should be
+        U, S, V = U[:, ::-1], S[::-1], V[:, ::-1]
+        V = V.T.conj()
 
         if not is_numpy:
             U = self.tensor(U, **ctx)
             S = self.tensor(S, **ctx)
             V = self.tensor(V, **ctx)
         return U, S, V
 
-    def truncated_svd(self, matrix, n_eigenvecs=None, **kwargs):
-        """Computes a truncated SVD on `matrix` using pytorch's SVD
+    def truncated_svd(self, matrix, n_eigenvecs=None):
```

Comment:
we could add a parameter `svd_fun="svd"` and carefully document it? This could then accept "numpy_svd", and other when applicable. That would probably be cleaner than what we have now.

Reply:
Why is `**kwargs` removed?

Reply:
I don't think it was actually used anywhere but we can add it back if it breaks things downstream.